pkg/controllers/dynakube/controller.go

```diff
@@ -322,6 +325,29 @@ func (controller *Controller) setupTokensAndClient(ctx context.Context, dk *dyna
 	return dynatraceClient, nil
 }
 
+func (controller *Controller) setOperatorImage(ctx context.Context, clt client.Client, apiReader client.Reader, dk *dynakube.DynaKube) {
+	if dk.Status.OperatorImage == "" {
```

Comment:
i'm not sure that this will work, if you set image once yes it wil. But when we change operator image after upgrade it can be outdated ... so we need different logic for it.

Reply:
done